### PR TITLE
Improving config file structure and removing modes.

### PR DESCRIPTION
**Pull request by VahidooX**

### CODE REVIEW

File: nemo/collections/nlp/data/text_classification/text_classification_descriptor.py [hunk 1]
```diff
@@ -12,43 +12,39 @@
...
 
 
 class TextClassificationDataDesc:
-    def __init__(self, data_dir: str, modes: List[str] = ['train', 'test', 'dev']):
+    def __init__(self, train_file: str, val_files: List[str] = []):
         """A descriptor class that reads all the data and calculates some stats of the data and also calculates the class weights to be used for class balancing
         Args:
             data_dir: the path to the data folder
-            modes: list of the modes to read, it can be from ["train", "test", "dev"] by default. It is going to look for the data files at {data_dir}/{mode}.tsv
+            modes: list of the modes to read, it can be from ["train", "test", "val"] by default. It is going to look for the data files at {data_dir}/{mode}.tsv
```
Comment (ekmb) [inline nemo/collections/nlp/data/text_classification/text_classification_descriptor.py:33]: please update the docsting

File: nemo/collections/nlp/data/text_classification/text_classification_descriptor.py [hunk 2]
```diff
@@ -12,43 +12,39 @@
...
     get_label_stats,
-    if_exist,
 )
 
 __all__ = ['TextClassificationDataDesc']
 
 
 class TextClassificationDataDesc:
-    def __init__(self, data_dir: str, modes: List[str] = ['train', 'test', 'dev']):
+    def __init__(self, train_file: str, val_files: List[str] = []):
```
Comment (ekmb) [inline nemo/collections/nlp/data/text_classification/text_classification_descriptor.py:29]: your validation in the text_classification model currently only supports a single file, maybe turn val_files into val_file and simply the code here?

File: examples/nlp/text_classification/conf/text_classification_config.yaml [hunk 3]
```diff
@@ -23,44 +23,42 @@ trainer:
   max_steps: null # precedence over max_epochs
   accumulate_grad_batches: 1 # accumulates grads every k batches
   amp_level: O0 # O1/O2 for mixed precision
+  precision: 32 # Should be set to 16 for O1 and O2
```
Comment (yzhang123) [inline examples/nlp/text_classification/conf/text_classification_config.yaml:26]: should be 16

File: examples/nlp/text_classification/conf/text_classification_config.yaml [hunk 4]
```diff
@@ -23,44 +23,42 @@ trainer:
...
     shuffle: true
     num_samples: -1
-    num_workers: 2
-    use_cache: false
-    drop_last: false
-    pin_memory: false
 
   validation_ds:
-    prefix: dev
+    file_name: ???
```
Comment (yzhang123) [inline examples/nlp/text_classification/conf/text_classification_config.yaml:49]: please put null

File: examples/nlp/text_classification/conf/text_classification_config.yaml [hunk 5]
```diff
@@ -23,44 +23,42 @@ trainer:
...
+    max_seq_length: 36
+    class_balancing: null # or weighted_loss
+    use_cache: false
+    num_workers: 2 # number of workers for data loaders
+    drop_last: false # drops the last last batch if it is smaller than the batch size
+    pin_memory: false # enables pin_memory feature of the data loaders
 
   train_ds:
-    prefix: train
+    file_name: ???
```
Comment (yzhang123) [inline examples/nlp/text_classification/conf/text_classification_config.yaml:43]: please put null

File: examples/nlp/text_classification/conf/text_classification_config.yaml [hunk 6]
```diff
@@ -75,8 +73,8 @@ model:
 
     sched:
```
Comment (ericharper) [inline examples/nlp/text_classification/conf/text_classification_config.yaml:74]: Optimizers and scheduler config has been simplified. See https://github.com/NVIDIA/NeMo/blob/e97aad2cd33bddb6937b9553b7d5b07b21f5fe1f/examples/asr/conf/config.yaml#L139

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 4 above)
Comment (VahidooX) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:49]: done

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 5 above)
Comment (VahidooX) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:43]: Shouldn't it be ??? to be necessary? Can user ignore training file and pass None?

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 6 above)
Comment (VahidooX) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:74]: Done

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 3 above)
Comment (VahidooX) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:26]: Is it OK to be 16 when amp_level is O0? I think they have the plan to drop amp_level and keep the precicion?

File: nemo/collections/nlp/data/text_classification/text_classification_descriptor.py (hunk 2 above)
Comment (VahidooX) [reply nemo/collections/nlp/data/text_classification/text_classification_descriptor.py:29]: But I should support both test and validation files.

File: nemo/collections/nlp/data/text_classification/text_classification_descriptor.py (hunk 1 above)
Comment (VahidooX) [reply nemo/collections/nlp/data/text_classification/text_classification_descriptor.py]: done.

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 5 above)
Comment (ekmb) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:45]: can we use just for eval if train_file is null?

File: examples/nlp/text_classification/conf/text_classification_config.yaml [hunk 7]
```diff
@@ -23,76 +23,67 @@ trainer:
...
-    drop_last: false
-    pin_memory: false
 
   language_model:
     pretrained_model_name: roberta-base
-    max_seq_length: 36
-    bert_checkpoint: null
-    bert_config: null
     tokenizer: nemobert # 'nemobert' or 'sentencepiece'
+    checkpoint_file: null # the path of a checkpoint file of a BERT-like model if a customized checkpoint is supposed to be used
```
Comment (ekmb) [inline examples/nlp/text_classification/conf/text_classification_config.yaml:59]: can we keep bert_checkpoint main? this name can be confused with the ckpt for the whole model

File: nemo/collections/nlp/models/text_classification/text_classification_model.py [hunk 8]
```diff
@@ -166,24 +170,34 @@ def setup_test_data(self, test_data_config: Optional[DictConfig]):
         self._test_dl = self.__setup_dataloader(cfg=test_data_config)
```
Comment (ekmb) [inline nemo/collections/nlp/models/text_classification/text_classification_model.py:171]: def setup_test_data should use setup_dataloader_from_config

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 7 above)
Comment (VahidooX) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:59]: done

File: nemo/collections/nlp/models/text_classification/text_classification_model.py (hunk 8 above)
Comment (VahidooX) [reply nemo/collections/nlp/models/text_classification/text_classification_model.py:171]: fixed.

File: nemo/collections/nlp/models/text_classification/text_classification_model.py [hunk 9]
```diff
@@ -47,23 +47,29 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
...
             tokenizer_name=cfg.language_model.tokenizer,
             pretrained_model_name=cfg.language_model.pretrained_model_name,
             vocab_file=cfg.language_model.vocab_file,
             tokenizer_model=cfg.language_model.tokenizer_model,
-            do_lower_case=cfg.language_model.do_lower_case,
+            do_lower_case=cfg.dataset.do_lower_case,
         )
 
+        self.model_cfg = cfg
+        # shared params for dataset and data loaders
```
Comment (ekmb) [inline nemo/collections/nlp/models/text_classification/text_classification_model.py:59]: we don't need this, after super.init() self._cfg == cfg

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 3 above)
Comment (yzhang123) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:26]: if you do precision 16 and O0 its still O0

File: nemo/collections/nlp/models/text_classification/text_classification_model.py (hunk 9 above)
Comment (VahidooX) [reply nemo/collections/nlp/models/text_classification/text_classification_model.py:59]: removed.

File: examples/nlp/text_classification/conf/text_classification_config.yaml (hunk 3 above)
Comment (VahidooX) [reply examples/nlp/text_classification/conf/text_classification_config.yaml:26]: Updated to 16.

**Pull request merged**